.github/workflows/060_generate_benchmark_weights.yml

```diff
@@ -0,0 +1,73 @@
+name: Generate benchmark weights
+
+on:
+  workflow_dispatch:
+
+jobs:
+  benchmark-weights:
+    runs-on: [self-hosted, tfchainrunner01]
+    container:
+      image: threefolddev/tfchain:4
+      env:
+        DEBIAN_FRONTEND: noninteractive
+        PATH: /root/.cargo/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/go/bin
+        RUSTUP_HOME: /root/.rustup
+        CARGO_HOME: /root/.cargo
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Chown user
+        run: |
+          chown -R $USER:$USER $GITHUB_WORKSPACE
+
+      - name: Cache build
+        uses: actions/cache@v3
+        timeout-minutes: 6
+        continue-on-error: true
+        with:
+          path: |
+            /root/.cargo/bin/
+            /root/.cargo/registry/index/
+            /root/.cargo/registry/cache/
+            /root/.cargo/git/db/
+            substrate-node/target/
+          key: ${{ runner.os }}-tfchain-build-cache-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: ${{ runner.os }}-tfchain-build-cache
+
+      - name: Build
+        run: |
+          cd substrate-node
+          cargo build --release --features runtime-benchmarks
+
+      - name: Remove old weight files
```

Comment:
Is it necessary to include this step? Are there any concerns about the generated files just overwriting existing ones?

Reply:
My concern is the custom script [benchmark.sh](https://github.com/threefoldtech/tfchain/blob/development/substrate-node/.maintain/benchmark.sh ) we’re using could become outdated when we introduce a new pallet, and that this step could delete the new pallet `weights.rs` without any guarantee that it will be regenerated. 

To ensure that both cleaning and generating these files won’t go out of sync, it might make sense to move this cleaning logic to the custom script. This way, if we change the script to adjust a path or a file name, the cleaning code would be in sight too to adjusted as well.

Alternatively, we can opt-out from using this script entirely and use the substrate node benchmark command directly in CI to generate these files for all extrinsic calls in all pallets, without being limited to only the pallets specified in the benchmark.sh script. This way, both logics will be in the same place and CI would continue to run after introducing new pallets.

Reply:
I opted for running a loop command directly in CI instead of [benchmark.sh](https://github.com/threefoldtech/tfchain/blob/development/substrate-node/.maintain/benchmark.sh ) script.
With this we ensure sync between removed and generated files.